src/manifest/deployment/index.ts

```diff
@@ -0,0 +1,378 @@
+import { ConfigurationListEntry } from '../ambient';
+import { ManifestDefinition, ResourceCollection, SubGenerator } from '../types';
+import { ListChoiceOptions, ListQuestion } from 'inquirer';
+import { DeploymentCollection, DeploymentDefinition } from './types';
+import { RepositoryCollection } from '../resource/Repository/types';
+import { EnvironmentCollection } from '../resource/Environment/types';
+import { Questions } from 'yeoman-generator';
+
+type DeploymentConfigurationEntry = ConfigurationListEntry<
+  DeploymentDefinition
+>;
+
+const strategies: Array<ListChoiceOptions> = [
+  {
+    name: 'Capistrano',
+    value: 'capistrano',
+  },
+  {
+    name: 'Artifact Repository',
+    value: 'artifact',
+  },
+];
+
+/**
+ * A Manifest sub-generator responsible for prompting and configuration of deployments.
+ *
+ * This generator is dependent on both the Environment and Repository sub-generators as well.
+ *
+ * @class Deployment
+ * @extends {SubGenerator}
```

Comment:
Most of these JSDoc annotations aren't needed - TypeScript knows from the `extends` declaration what this is. (This also applies to `@export` and `@memberof`.)

```suggestion
```